### PR TITLE
avoid creating a promise

### PR DESCRIPTION
**Pull request by mcollina**
this pr:

```
$ autocannon -c 100 -d 5 -p 10 localhost:3000
Running 5s test @ http://localhost:3000
100 connections with 10 pipelining factor

Stat         Avg     Stdev   Max
Latency (ms) 3.62    11.76   291.03
Req/Sec      27051.2 1275.73 28229
Bytes/Sec    4.42 MB 233 kB  4.63 MB

135k requests in 5s, 22.2 MB read
```

master:

```
$ autocannon -c 100 -d 5 -p 10 localhost:3000
Running 5s test @ http://localhost:3000
100 connections with 10 pipelining factor

Stat         Avg    Stdev   Max
Latency (ms) 3.78   11.99   250.12
Req/Sec      25912  1670.52 27637
Bytes/Sec    4.3 MB 277 kB  4.53 MB

130k requests in 5s, 21.2 MB read
```

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by allevo** [comment]
I don't understand the benchmarks. `Before` is 135k, `After` is 130k. What have I lost?

**Comment by mcollina** [comment]
I’ve just inverted the labels :). Fixed now

File: lib/handleRequest.js [hunk 1]
```diff
@@ -85,16 +85,23 @@ function preHandlerCallback (err, reply) {
...
+            reply.sent = false
+            reply._isError = true
+            reply.send(err)
+          }
         } else if (reply.sent === false) {
           reply.res.log.error(new Error(`Promise may not be fulfilled with 'undefined' when statusCode is not 204`))
         }
-      })
-      .catch((err) => {
+      }, function (err) {
```
Comment (StarpTech) [inline lib/handleRequest.js]: Only a nit, could you use `catch` it's more verbose.

Comment (nwoltman) [reply lib/handleRequest.js]: The whole point of this PR is to remove `catch` to avoid creating another promise object.

Comment (StarpTech) [reply lib/handleRequest.js]: Oh 😸 , you're right I haven't seen it.

**Comment by StarpTech** [comment]
The same for https://github.com/fastify/fastify/blob/master/lib/reply.js#L295 ?

**Comment by mcollina** [comment]
@StarpTech that's not sitting in a major hot path, so I would not bother. But yes, you are right it might improve to have it there as well.

**Pull request merged**